plugin/storage/s3/s3.go

```diff
@@ -83,7 +84,9 @@ func (client *Client) UploadFile(ctx context.Context, filename string, fileType
 	link := uploadOutput.Location
 	// If url prefix is set, use it as the file link.
 	if client.Config.URLPrefix != "" {
-		link = fmt.Sprintf("%s/%s%s", client.Config.URLPrefix, filename, client.Config.URLSuffix)
+		parts := strings.Split(filename, "/")
```

Comment:
The use case here is the filename variable doesn't only contain filename if we configure a storage path like "{year}/{month}/{timestamp}_{filename}" (which is suggested in the placeholder of this field) in Memos settings.

In this case, the "/" in "2024/01/测试.png" shouldn't be encoded.

Reply:
To prevent some users from putting filename in front of it, such as `{year}/{month}/{filename}/{timestamp}`. We can't make all users using the standard format. So I think we should escape all parts.